model/DataObject.php

```diff
@@ -1744,63 +1756,80 @@ public function getManyManyComponents($componentName, $filter = "", $sort = "",
 	}
 
 	/**
+	 * @deprecated 4.0 Method has been renamed to hasOne()
+	 * @param string $component
+	 * @return array|null
+	 */
+	public function has_one($component = null) {
+		Deprecation::notice('4.0', 'Please use hasOne() instead');
```

Comment:
to save getting multiple deprecation notices, perhaps ths should check `if $compontent !== null` and throwing a "use hasOneComponent($component)" notice instead?

At the moment a dev would see two contradictory notices otherwise...